### PR TITLE
FIX user notification for ticket assigned on creation

### PR DESCRIPTION
**Pull request by BenjaminFlr**
When an user is assigned to a ticket directly during the creation, there is no email notification whereas there is one when the user is assigned AFTER the creation.

I propose this little change in order to fix it and notify an user even when he's assigned during the ticket creation.

### CODE REVIEW

File: htdocs/ticket/class/ticket.class.php [hunk 1]
```diff
@@ -568,6 +567,13 @@ public function create($user, $notrigger = 0)
 				$this->id = $this->db->last_insert_id(MAIN_DB_PREFIX."ticket");
 			}
 
+			// Assign user if defined
```
Comment (eldy) [inline htdocs/ticket/class/ticket.class.php:570]: The create method is a low level CRUD method, so we should not have other business event and trigger inside the create.
Call of the assignUser should be done after calling the create by the parent page orchestrating the create call.

Comment (altairis-benjamin) [reply htdocs/ticket/class/ticket.class.php:570]: Thanks for the feedback. I didn't know what was the best way to do this.

I'll make the required changes.

**Comment by BenjaminFlr** [comment]
I made the changes.
@eldy : it should now do the trick.

**Comment by BenjaminFlr** [comment]
Any update on here?

**Comment by BenjaminFlr** [comment]
@eldy all the pre-commit tests are now ok :)

**Comment by BenjaminFlr** [comment]
Little up here @eldy :)

File: htdocs/core/triggers/interface_50_modTicket_TicketEmail.class.php [hunk 2]
```diff
@@ -190,6 +170,35 @@ public function runTrigger($action, $object, User $user, Translate $langs, Conf
...
+							if (!empty($conf->global->TICKET_DISABLE_MAIL_AUTOCOPY_TO)) {
+								$old_MAIN_MAIL_AUTOCOPY_TO = $conf->global->MAIN_MAIL_AUTOCOPY_TO;
+								$conf->global->MAIN_MAIL_AUTOCOPY_TO = '';
+							}
+
+							if (!empty($sendto)) {
+								$this->composeAndSendAssigneeMessage($sendto, $subject_assignee, $body_assignee, $see_ticket_assignee, $object, $langs);
+							}
+
+							if (!empty($conf->global->TICKET_DISABLE_MAIL_AUTOCOPY_TO)) {
```
Comment (eldy) [inline htdocs/core/triggers/interface_50_modTicket_TicketEmail.class.php:192]: Use of $conf->global-> is forbidden.
Must use getDolGlobalString or getDolGlobalInt

File: htdocs/core/triggers/interface_50_modTicket_TicketEmail.class.php [hunk 3]
```diff
@@ -461,4 +470,62 @@ private function composeAndSendCustomerMessage($sendto, $base_subject, $body, $s
...
+	{
+		global $conf, $user;
+
+		// Init to avoid errors
+		$filepath = array();
+		$filename = array();
+		$mimetype = array();
+
+		// Send email to assigned user
+		$subject = '['.$conf->global->MAIN_INFO_SOCIETE_NOM.'] '.$langs->transnoentities($base_subject);
```
Comment (eldy) [inline htdocs/core/triggers/interface_50_modTicket_TicketEmail.class.php:495]: Old code was using
'['.$mysoc->name.']'
why changing into 
$conf->global->MAIN_INFO_SOCIETE_NOM  ?

File: htdocs/core/triggers/interface_50_modTicket_TicketEmail.class.php (hunk 2 above)
Comment (altairis-benjamin) [reply htdocs/core/triggers/interface_50_modTicket_TicketEmail.class.php:192]: It was used a few lines ahead so I didn't think to use the function. Now fixed.

File: htdocs/core/triggers/interface_50_modTicket_TicketEmail.class.php (hunk 3 above)
Comment (altairis-benjamin) [reply htdocs/core/triggers/interface_50_modTicket_TicketEmail.class.php]: Because it wasn't passed as global, I just changed it.

**Comment by W1W1-M** [comment]
Hello,
This fix would be a great "little" addition for ticketing system !

**Comment by eldy** [comment]
Replaced by #31928 that has no conflict.

**Pull request closed without merging**